### PR TITLE
NIST NVD (independent publisher)

### PR DESCRIPTION
**Pull request by paulculmsee**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

Please note that we are experiencing a significant delay due to a high volume of connector submissions. It may take 1-2 weeks for us to provide comments for your connector.

- [x ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
image
image
image

- [x ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI. NA. Description updates

image

- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: independent-publisher-connectors/NIST National Vulnerability Database/readme.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+* Each API Key is associated with a single email address. If an email address is used to request an additional API key, clicking the single-use hyperlink will invalidate the key previously associated with that email address. The key will not be invalidated if the email is used to request another key, but the link is not opened. There is no process for retrieving a forgotten key.
+
+* The rate limit with an API key is 100 requests in a rolling 60 second window.
+
+* The best practice for making requests within the rate limit is to use the modified date parameters. No more than once every two hours, automated requests should include a range where modStartDate equals the time of the last CVE or CPE received and modEndDate equals the current time. Enterprise scale development should enforce this approach through a single requestor to ensure all users are in sync and have the latest CVE and CPE information. It is also recommended that users "sleep" their scripts for six seconds between requests.
+
+```
+paconn create --api-def apiDefinition.swagger.json --api-prop apiProperties.json.
```
Comment (v-eisljami) [inline independent-publisher-connectors/NIST National Vulnerability Database/readme.md:37]: Readme file doesn't contain Obtaining Credentials section. Even if there are no credentials this section is mandatory as defined in this [template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md). Please review the template and check your readme. Also, I think that this line got in the readme by mistake. If so, please remove it.

File: independent-publisher-connectors/NIST National Vulnerability Database/readme.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Get a collection of CVE. The Common Vulnerabilities and Exposures (CVE) program is a dictionary or glossary of vulnerabilities that have been identified for specific code bases, such as software applications or open libraries 
+
+### Retrieve CPE information
+Get a collection of CPE. The Official CPE Dictionary, is a searchable repository of hardware and software products maintained by the National Vulnerability Database (NVD)
+
+## API Documentation
+https://nvd.nist.gov/developers
+
+## Known Issues and Limitations
+* FIltering by date is very specific in terms of format.  It is unlikely you need to filter based on time so I suggest you specify 000 for subseconds and use hh:mm for UTC offset. Eg. For Power Automate flows, this is an example of correctly specifying a date with no UTC offset. formatDateTime(utcNow(), 'yyyy-MM-ddTHH:mm:ss:000 UTC+00:00')
```
Comment (v-eisljami) [inline independent-publisher-connectors/NIST National Vulnerability Database/readme.md:26]: Readme file should be free from grammar and spelling errors. Please review the file. We recommend using some grammar/spelling checking tools.

File: independent-publisher-connectors/NIST National Vulnerability Database/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
+                                "items": {
+                                  "type": "object",
+                                  "properties": {
+                                    "operator": {
+                                      "type": "string",
+                                      "description": "Logical Operator"
+                                    },
+                                    "children": {
+                                      "type": "array",
+                                      "items": {},
```
Comment (v-eisljami) [inline independent-publisher-connectors/NIST National Vulnerability Database/apiDefinition.swagger.json:213]: What is the type of items in the array? All schemas must be well defined.

Comment (paulculmsee) [reply independent-publisher-connectors/NIST National Vulnerability Database/apiDefinition.swagger.json:213]: This is a recursive schema where the data type is the same as the parent (node). Hence I left it to implementers here as its impossible to predict the depth

**Comment by sriyen-msft** [comment]
@paulculmsee Can you please resolve the conflicts to approve the PR

**Comment by sriyen-msft** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hi @paulculmsee, I hope you are doing well. I want to follow up with you and let you know that your connector is in production deployment. this process will complete within 4-6 weeks.
Please let me know if you have any questions.
Thank you very much.

**Review by sriyen-msft — APPROVED** [review]

**Comment by sriyen-msft** [comment]
@paulculmsee Can you please resolve the branch conflicts to merge the PR?

**Comment by sriyen-msft** [comment]
@paulculmsee Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, Thanks.

**Comment by sriyen-msft** [comment]
@paulculmsee Please sign the CLA license

**Comment by sriyen-msft** [comment]
@paulculmsee Apologies for the delay in response. 

Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, Thanks.

**Review by sriyen-msft — APPROVED** [review]

**Review by sriyen-msft — CHANGES_REQUESTED** [review]
@paulculmsee Seems that you are updating the NIST NVD connector, but there are other connector files in the PR, Can you please remove them to move forward with your connector update in the certification process. Thanks

**Comment by Amjed-Ayoub** [comment]
Hi @paulculmsee I hope you are doing well. I am following up with you to see how things are going with updating your pull request to remove other connector files.
Please let us know if you have any questions.
We look forward to getting your connector certified.
Thank you very much.

**Comment by paulculmsee** [comment]
Sorry mate I don't know how it got so messed up, so I deleted that branch and resubmitted a new PR

**Pull request closed without merging**